### PR TITLE
Update type `__package_update_index`

### PR DESCRIPTION
**Pull request by zhaostu**
* Fixed the issue where `__package_update_index` won't run with package managers other than apt.
* Support `--maxage` parameter for Arch Linux.
* Refactored the code to handle os and package manager type correctly.

### CODE REVIEW

**Comment by darko-poljak** [comment]
@telmich @asteven Can you too take a look at this one? Thanks!

File: cdist/conf/type/__package_update_index/explorer/currage [hunk 1]
```diff
@@ -17,18 +17,24 @@
 # You should have received a copy of the GNU General Public License
 # along with cdist. If not, see <http://www.gnu.org/licenses/>.
 
-os="$("$__explorer/os")"
+type="$($__type_explorer/type)"
```
[the remark this answers is not in the document]
Comment (zhaostu) [reply cdist/conf/type/__package_update_index/explorer/currage]: Please see here:

https://www.nico.schottelius.org/software/cdist/man/latest/cdist-explorer.html

[the remark this answers is not in the document]
Comment (asteven) [reply cdist/conf/type/__package_update_index/explorer/currage]: __type_explorer exists and is used correctly

**Comment by asteven** [comment]
@zhaostu thank you for your contribution!

The patch is very cleanly done. I like the code and style.

Assuming that the behavior of the type has not changed and for existing users/consumers it works the same as it did before and does the same thing I'm +1.

File: cdist/conf/type/__package_update_index/explorer/currage (hunk 1 above)
[the remark this answers is not in the document]
Comment (darko-poljak) [reply cdist/conf/type/__package_update_index/explorer/currage]: @zhaostu @asteven Sorry! You are right, I missed it.

**Comment by zhaostu** [comment]
@asteven Thanks! I did test this on my Arch Linux box with several different settings. It's a little bit scary though as there are no automated tests (and caused the problem in the first place). I wonder how cdist can be tested.

**Comment by darko-poljak** [comment]
@zhaostu @asteven Well, the best testing environment is the production ;)

I am merging this PR, @zhaostu  thanks for your contribution.

**Pull request merged**